packages/flutter/lib/src/widgets/automatic_keep_alive.dart

```diff
@@ -80,29 +80,61 @@ class _AutomaticKeepAliveState extends State<AutomaticKeepAlive> {
     handle.addListener(_handles[handle]);
     if (!_keepingAlive) {
       _keepingAlive = true;
-      // We use Element.visitChildren rather than context.visitChildElements
-      // because we might be called during build, and context.visitChildElements
-      // verifies that it is not called during build. Element.visitChildren does
-      // not, instead it assumes that the caller will be careful. (See the
-      // documentation for these methods for more details.)
-      //
-      // Here we know it's safe because we just received a notification, which
-      // we wouldn't be able to do if we hadn't built our child and its child --
-      // our build method always builds the same subtree and it always includes
-      // the node we're looking for (KeepAlive) as the parent of the node that
-      // reports the notifications (NotificationListener).
-      //
-      // (We're only going down one level, to get our direct child.)
-      final Element element = context;
-      element.visitChildren((Element child) {
-        assert(child is ParentDataElement<SliverMultiBoxAdaptorWidget>);
-        final ParentDataElement<SliverMultiBoxAdaptorWidget> childElement = child;
-        childElement.applyWidgetOutOfTurn(build(context));
-      });
+      final ParentDataElement<SliverMultiBoxAdaptorWidget> childElement = _getChildElement();
+      if (childElement != null) {
+        // If the child already exists, update it synchronously.
+        _updateParentDataOfChild(childElement);
+      } else {
+        // If the child doesn't exist yet, we got called during the very first
+        // build of this subtree. Wait until the end of the frame to update
+        // the child when the child is guaranteed to be present.
+        SchedulerBinding.instance.addPostFrameCallback((_) {
```

Comment:
we should be explicit about argument names and types even if we don't use them, that way people editing or copying the code don't have to look up what it is